traj/src/traj/fanuc_limits.py

```diff
@@ -0,0 +1,134 @@
+
+# Copyright (c) 2020, G.A. vd. Hoorn
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author G.A. vd. Hoorn
+
+import numpy as np
+
+from scipy.interpolate import interp2d
+
+
+def gen_limit_interpolation_func(no_load_thresh, max_load_thresh, payload_max,
+                                 cart_vmax=4e3, num_steps=20,
+                                 interp_func='linear'):
+    """Generates a function which can be used to lookup the scaled (positive)
+    limit for a joint based on the NO and MAX load threshold tables, payload
+    and Cartesian velocity of the TCP of a Fanuc with J519.
+
+    The returned function object wraps the function returned by a call to
+    scipy.interp2d(..), taking in current Cartesian velocity of the TCP
+    (in mm/s) and current weight of the payload (in Kg) and reeturns the 2D
```

Comment:
Could you change the cartesian velocity to m/s?

Reply:
Nitpicky detail question: Is this cartesian velocity the velocity at the previous timestep, or is it the cartesian velocity that we expect to be achieved at the next time step?

Reply:
> Could you change the cartesian velocity to m/s?

well, Fanuc specifies velocities in mm/s, and seeing as this is Fanuc specific, I was thinking to keep using their units.

The default values for the threshold tables will also be in degrees/s (etc).

> Nitpicky detail question: Is this cartesian velocity the velocity at the previous timestep, or is it the cartesian velocity that we expect to be achieved at the next time step?

According to the information I have it's the current Cartesian velocity (as that determines what is possible the next timestep). But I'll check again.

Reply:
This is fanuc specific, but the trajectory planning functions that will call this function shouldn't have to worry about unit conversions. So maybe it does make sense to pass the max cartesian value in as mm/sec (since those may just be read in from some fanuc-specific file), but i think that the function that this function returns (the actual interpolation function) should operate on m/s for cartesian velocity.

Reply:
Fair enough.